woocommerce/woocommerce-ios
# Coupons: Add new view for enabling analytics

Closes: #6360 


### Description

This PR adds a new screen to provide an explanation and an action button to enable WooCommerce Analytics from within the app. This flow is needed to address the issue with loading the total discounted amount on the coupon details screen since we're using a `wc-analytics` endpoint for this.

A few things to note:
- We have 2 UIKit flows to handle on this screen: navigation to support and displaying notices for successful or failed requests. Both these flows require a view controller to present from; while the enable analytics screen is a modal presented from the coupon detail screen in SwiftUI. The solution is to send the presenting view controller to the `EnableAnalyticsView`, which essentially is the hosting controller of the coupon details screen. This workaround doesn't look very nice, so please feel free to suggest another solution if possible.
- The API request to toggle analytics settings doesn't work very consistently: if the value in the request is different from the current setting value, the request always fails on first try and succeeds on the second time. I'm not sure yet where to file the issue (probably [woocommerce/woocommerce-admin](https://github.com/woocommerce/woocommerce-admin)?) But for now, I've added a workaround by allowing retrying the request once to avoid the false failure.

### Testing instructions

- Please make sure that your test stores has coupons enabled (in WP Admin > WC Settings > General > Enable Coupons), has at least one coupon set up (in WP Admin > Marketing > Coupons) and applied to at least an order.
- Disable WC Analytics in WP Admin > WC Settings > Advanced > Features > Analytics.
- Open the app, navigate to Menu > Coupons > select the coupon that has been applied to at least an order.
- On the coupon details screen, notice that there's an orange (i) icon next to "Amount" after loading the total discounted amount fails. Tapping on that, notice that a new screen for enabling analytics is presented.
- Tap on Contact Support, notice that the support flow is presented properly.
- Tap on Not now, notice that the screen is dismissed.
- Tap on Enable analytics, notice that the screen should be dismissed if the request succeeds. The coupon details screen should reload its Amount field.
- Try again with the internet connection off. Tap on Enable analytics should not dismiss the screen and display a notice about the error instead.

### Screenshots


https://user-images.githubusercontent.com/5533851/158966104-a40efcbd-10be-4592-9f98-b3dac8173a37.mp4



---
